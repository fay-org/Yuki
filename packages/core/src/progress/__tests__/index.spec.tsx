import { mount } from '@vue/test-utils'
import Progress from '..'

describe('Progress', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Progress)
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('fixed', () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <Progress fixedTop />
            <Progress fixedBottom />
          </>
        )
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('color', () => {
    const wrapper = mount(Progress, {
      props: {
        colors: {
          20: 'red',
          40: 'blue',
          60: 'green',
          70: 'pink'
        },
        value: 45
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(
      wrapper
        .find('.fect-progress')
        .attributes('style')
        ?.split(';')
        .find((v) => v.includes('--progress-inner-bg-color:'))
        ?.replace(/\s/, '')
    ).toEqual('--progress-inner-bg-color: green')
  })
})
