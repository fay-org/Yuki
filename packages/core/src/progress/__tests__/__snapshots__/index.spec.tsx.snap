// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress color 1`] = `
"<div class="fect-progress" style="--progress-radius: 6px; --progress-bg-color: #eaeaea; --progress-inner-width: 45%; --progress-inner-bg-color: green; --progress-fixed-bottom: unset; --progress-fixed-top: unset; --progress-width: 100%; --progress-height: calc(0.625 * 16px); --progress-pt: 0; --progress-pr: 0; --progress-pb: 0; --progress-pl: 0; --progress-mt: 0; --progress-mr: 0; --progress-mb: 0; --progress-ml: 0;">
  <div class="fect-progress__inner fect-progress__inner--default" title="45%"></div><progress max="100" value="45"></progress>
</div>"
`;

exports[`Progress fixed 1`] = `
"<div class="fect-progress fect-progress--fixed" style="--progress-radius: 6px; --progress-bg-color: #eaeaea; --progress-inner-width: 0%; --progress-inner-bg-color: #000; --progress-fixed-bottom: unset; --progress-fixed-top: 0; --progress-width: 100%; --progress-height: calc(0.625 * 16px); --progress-pt: 0; --progress-pr: 0; --progress-pb: 0; --progress-pl: 0; --progress-mt: 0; --progress-mr: 0; --progress-mb: 0; --progress-ml: 0;">
  <div class="fect-progress__inner fect-progress__inner--default" title="0%"></div><progress max="100" value="0"></progress>
</div>
<div class="fect-progress fect-progress--fixed" style="--progress-radius: 6px; --progress-bg-color: #eaeaea; --progress-inner-width: 0%; --progress-inner-bg-color: #000; --progress-fixed-bottom: 0; --progress-fixed-top: unset; --progress-width: 100%; --progress-height: calc(0.625 * 16px); --progress-pt: 0; --progress-pr: 0; --progress-pb: 0; --progress-pl: 0; --progress-mt: 0; --progress-mr: 0; --progress-mb: 0; --progress-ml: 0;">
  <div class="fect-progress__inner fect-progress__inner--default" title="0%"></div><progress max="100" value="0"></progress>
</div>"
`;
